tests/install_upgrade_operators/console_cli_download/conftest.py

```diff
@@ -45,24 +69,61 @@ def internal_fqdn(admin_client, hco_namespace):
 
 
 @pytest.fixture()
-def non_internal_fqdns(all_virtctl_urls, internal_fqdn):
+def non_internal_fqdns(all_virtctl_urls_scope_function, internal_fqdn):
     """
     Get URLs containing FQDN that is not matching the cluster's route
 
     Returns:
         list: list of all non-internal FQDNs
     """
-    return [virtctl_url for virtctl_url in all_virtctl_urls if f"//{internal_fqdn}" not in virtctl_url]
+    return [virtctl_url for virtctl_url in all_virtctl_urls_scope_function if f"//{internal_fqdn}" not in virtctl_url]
 
 
 @pytest.fixture()
-def downloaded_and_extracted_virtctl_binary_for_os(request, all_virtctl_urls, tmpdir):
+def downloaded_and_extracted_virtctl_binary_for_os(request, all_virtctl_urls_scope_function, tmpdir):
     """
     This fixture downloads the virtctl archive from the provided OS, and extracts it to a temporary dir
     """
     return get_and_extract_file_from_cluster(
         system_os=request.param.get("os"),
-        urls=all_virtctl_urls,
+        urls=all_virtctl_urls_scope_function,
         dest_dir=tmpdir,
         machine_type=request.param.get("machine_type"),
     )
+
+
+@pytest.fixture(scope="class")
+def ingress_resource(admin_client):
+    return Ingress(client=admin_client, name="cluster", ensure_exists=True)
```

Comment:
same, only one action in fixture and used only in one place.. do you really need it?